cdap-ui/graphql/resolvers/type/programRecordTypeResolver.js

```diff
@@ -14,6 +14,35 @@
  * the License.
  */
 
+var urlHelper = require('../../../server/url-helper'),
+  cdapConfigurator = require('../../../server/cdap-config.js'),
+  resolversCommon = require('../resolvers-common.js');
+
+let cdapConfig;
+cdapConfigurator.getCDAPConfig().then(function(value) {
+  cdapConfig = value;
+});
+
+const runsResolver = async (programType, parent, args, context) => {
+  const namespace = context.namespace;
+  const name = parent.app;
+  const programId = parent.name;
+  const options = resolversCommon.getPOSTRequestOptions();
+  options.headers.Authorization = context.auth;
+  options['url'] = urlHelper.constructUrl(cdapConfig, `/v3/namespaces/${namespace}/runs`);
+  options['body'] = [
+    {
+      appId: name,
+      programType: programType,
+      programId: programId,
+    },
+  ];
+
+  const runInfo = await resolversCommon.requestPromiseWrapper(options);
+
+  return runInfo[0].runs;
```

Comment:
this is unsafe. runInfo may not have index 0

Reply:
Done